inc/sainat.h

```diff
@@ -0,0 +1,653 @@
+/**
+ * Copyright (c) 2014 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    sainat.h
+ *
+ * @brief   This module defines SAI NAT (Network Address Translation) spec
+ */
+
+#if !defined (__SAINAT_H_)
+#define __SAINAT_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAINAT SAI - Network Address Translation (NAT) specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief NAT Type
+ */
+typedef enum _sai_nat_type_t
+{
+    /** No NAT */
+    SAI_NAT_TYPE_NONE,
+
+    /** Source NAT */
+    SAI_NAT_TYPE_SOURCE_NAT,
+
+    /** Destination NAT */
+    SAI_NAT_TYPE_DESTINATION_NAT,
+
+    /** Double NAT */
+    SAI_NAT_TYPE_DOUBLE_NAT,
+
+} sai_nat_type_t;
+
+/**
+ * @brief NAT Entry Attributes for Match
+ */
+typedef enum _sai_nat_entry_attr_t
+{
+
+    /**
+     * @brief Start of Attributes
+     */
+    SAI_NAT_ENTRY_ATTR_START,
+
+    /**
+     * @brief NAT Type defined in sai_nat_type_t
+     * @type sai_nat_type_t
+     * @flags CREATE_AND_SET
+     * @default SAI_NAT_TYPE_NONE
+     */
+    SAI_NAT_ENTRY_ATTR_NAT_TYPE  = SAI_NAT_ENTRY_ATTR_START,
+
+    /**
+     * @brief NAT Inside Zone ID
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_NAT_ENTRY_ATTR_FROM_ZONE_ID,
```

Comment:
What is the purpose of the zones in NAT entry?
According to the pipeline, we don't distinguish between the specific zones, but only when IN_ZONE!=OUT_ZONE, where IN_ZONE is an attribute of IRIF, and the OUT_ZONE - of ERIF.

Reply:
This we have discussed and closed in SAI pipeline